### ISSUE TITLE
community.docker.docker_container incorrect return values

### ISSUE THREAD

**Issue by agowa338** [ansible-collections/community.docker#177]
##### SUMMARY

community.docker.docker_container sometimes doesn't return the defined return values.

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME
community.docker.docker_container

##### ANSIBLE VERSION
```paste below
ansible 2.10.11
  python version = 3.9.6 (default, Jun 30 2021, 10:22:16) [GCC 11.1.0]
```

##### COLLECTION VERSION

```paste below

```

##### CONFIGURATION

```paste below

```

##### OS / ENVIRONMENT
ArchLinux

##### STEPS TO REPRODUCE
I'm not quite sure, but as far as I've seen it so far, it occurs if the command being executed fails.
```yaml
- name: ansible-certbot | bootstrap container
  community.docker.docker_container:
    name: '{{ ansible_certbot_container_name }}'
    state: started
    pull: true
    purge_networks: true
    networks_cli_compatible: yes
    network_mode: default
    container_default_behavior: compatibility
    image: '{{ ansible_certbot_image }}'
    ports: >-
      {{ '[' + (ansible_certbot_expose_port) | ternary(' "' +
      ansible_certbot_host_ip + ':' ~ ansible_certbot_host_port +':80"','') +
      ']' }}
    networks: '{{ ansible_certbot_docker_networks }}'
    volumes:
      - '{{ ansible_certbot_project_directory }}/etc:/etc/letsencrypt:rw'
      - '{{ ansible_certbot_project_directory }}/lib:/var/lib/letsencrypt:rw'
      - '{{ ansible_certbot_project_directory }}/log:/var/log/letsencrypt:rw'
    detach: false
    tty: true
    command: >-
      certonly --standalone -d {{ item }} -m "{{ ansible_certbot_contact_mail }}"
      --agree-tos --keep-until-expiring --preferred-challenges http-01
      --http-01-address 0.0.0.0 -vvvv {% if ansible_certbot_testing %}
      --test-cert {% endif %}
      --staple-ocsp
      --key-type rsa --rsa-key-size 4096
  register: container_res
  changed_when:
    - >-
      'Congratulations! Your certificate and chain have been saved at:' in
      container_res.container.Output
  failed_when:
    - container_res is failed
    - >-
      'The following errors were reported by the server:' in
      container_res.container.Output
  with_items:
    - incorrect-domain.de
```

##### EXPECTED RESULTS
`.container` should always be populated.

##### ACTUAL RESULTS
container_res contains:
```paste below
{
  "ansible_loop_var":"item",
  "changed":false,
  "item":"incorrect-domain.de",
  "msg":"Root logging level set at -10\r\nSaving debug log to /var/log/letsencrypt/letsencrypt.log(...)",
  "status":1
}
```

E.g. the attribute `.msg` should actually be at `.container.Output`

**Comment by felixfontein** [comment]
I created a fix in #178. It corrects the documentation (`container` information is usually only there in case of success, with special case `state=present` and `detached=false`, where it is now also present when `status != 0`). Please check out whether it solves your problem.

**Comment by agowa338** [comment]
I've tested it right now and it works as expected. Thanks for the quick fix.
